modules/era/lua_dynamis/globals/era_dynamis_spawning.lua

```diff
@@ -1935,6 +1974,12 @@ xi.dynamis.setNightmareStats = function(mob)
 
         -- Add Check After Calcs
         mob:setMobMod(xi.mobMod.CHECK_AS_NM, 2)
+
+        if mob:getZone():getID() == xi.zone.DYNAMIS_TAVNAZIA then
+            mob:setRoamFlags(xi.roamFlag.SCRIPTED)
```

Comment:
should the worms be allowed to move?  if so they'll need worm roam flag

Do the worms cast on players if they have hate within melee range?  This is also controlled via the worm roam flag

Reply:
I believe you are correct, we can add that to the onSpawnNightmareWormfunction.